File: DeepLearningServer/Models/ModelRecord.cs
using System.ComponentModel.DataAnnotations;

namespace DeepLearningServer.Models
{
    public class ModelRecord
    {
        public int Id { get; set; }

        [Required]
        public string ModelName { get; set; } = null!;

        [Required]
        public string Status { get; set; } = "saved"; // "deployed", "saved", "error" 등 상태 관리

        public string? ClientPath { get; set; } // 클라이언트 저장 경로
        public string? ServerPath { get; set; } // 딥러닝 서버 내부 저장 경로

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // 🔹 AdmsProcessType을 참조 (라지/미들 모델 구분)
        [Required]
        public int AdmsProcessTypeId { get; set; }
        public virtual AdmsProcessType AdmsProcessType { get; set; } = null!;

        // 🔹 TrainingRecord를 참조 (훈련 기록과 연결)
        public int? TrainingRecordId { get; set; }
        public virtual TrainingRecord? TrainingRecord { get; set; }
    }
}

File: deeplearningserver-python/models/model_record.py
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from models.base import Base


class ModelRecord(Base):
    __tablename__ = "ModelRecords"

    id: Mapped[int] = mapped_column("Id", Integer, primary_key=True)
    model_name: Mapped[str] = mapped_column("ModelName", String(255), nullable=False)
    status: Mapped[str] = mapped_column("Status", String(50), nullable=False, default="saved")
    client_path: Mapped[str | None] = mapped_column("ClientPath", String(500))
    server_path: Mapped[str | None] = mapped_column("ServerPath", String(500))
    created_at: Mapped[datetime] = mapped_column("CreatedAt", DateTime, default=datetime.now)
    updated_at: Mapped[datetime | None] = mapped_column("UpdatedAt", DateTime)
    adms_process_type_id: Mapped[int] = mapped_column(
        "AdmsProcessTypeId", ForeignKey("AdmsProcessTypes.Id"), nullable=False
    )
    training_record_id: Mapped[int | None] = mapped_column(
        "TrainingRecordId", ForeignKey("TrainingRecords.Id")
    )

    adms_process_type = relationship("AdmsProcessType", back_populates="model_records")
    training_record = relationship("TrainingRecord", back_populates="model_records")
